src/GovUk.Education.ExploreEducationStatistics.Data.Model/Repository/LocationRepository.cs

```diff
@@ -86,49 +62,36 @@ public LocationRepository(StatisticsDbContext context)
                 ? new Dictionary<GeographicLevel, List<Func<Location, ILocationAttribute>>>()
                 : MapLocationAttributeSelectors(hierarchies);
 
-            var locationsWithGeographicLevels = observations
+            var locationIds = observations
```

Comment:
Can this instead now be:

```
var locations = await observations
    .AsNoTracking()
    .Select(observation => observation.Location)
    .Distinct()
    .ToListAsync();
```

or 

```
var locations = await observations
    .AsNoTracking()
    .Include(observation => observation.Location)
    .Select(observation => observation.Location)
    .Distinct()
    .ToListAsync();
```

instead of firstly obtaining the ids and then selecting locations by those ids?  Don't know if one is faster than the other?

Reply:
I checked both suggestions and they both translate to exactly the same SQL.

I've tidied this up a bit to make it a bit more readable:

```
SELECT DISTINCT
                l.Id,
                l.Country_Code,
                l.Country_Name,
                l.EnglishDevolvedArea_Code,
                l.EnglishDevolvedArea_Name,
                l.GeographicLevel,
                l.Institution_Code,
                l.Institution_Name,
                l.LocalAuthorityDistrict_Code,
                l.LocalAuthorityDistrict_Name,
                l.LocalAuthority_Code,
                l.LocalAuthority_Name,
                l.LocalAuthority_OldCode,
                l.LocalEnterprisePartnership_Code,
                l.LocalEnterprisePartnership_Name,
                l.MayoralCombinedAuthority_Code,
                l.MayoralCombinedAuthority_Name,
                l.MultiAcademyTrust_Code,
                l.MultiAcademyTrust_Name,
                l.OpportunityArea_Code,
                l.OpportunityArea_Name,
                l.ParliamentaryConstituency_Code,
                l.ParliamentaryConstituency_Name,
                l.PlanningArea_Code,
                l.PlanningArea_Name,
                l.Provider_Code,
                l.Provider_Name,
                l.Region_Code,
                l.Region_Name,
                l.RscRegion_Code,
                l.School_Code,
                l.School_Name,
                l.Sponsor_Code,
                l.Sponsor_Name,
                l.Ward_Code,
                l.Ward_Name
FROM Observation AS o
         INNER JOIN Location AS l ON o.LocationId = l.Id
WHERE o.SubjectId = '<Subject Id>'
```

I've gone for the option without the `.Include`.

I can't see exactly where in the docs the `.Select` overrides the need for the `.Include` but the tests in `LocationRepositoryTests` should fail if we needed it and they don't.

I checked the performance in my local database when compared with the old two stage approach and found this was slightly faster. With the Subject with the largest number of rows in Prod I didn't find any performance problem.

I could test this further but I think we agree that the JOIN is going to be faster especially where there are a large number of Locations.